compose/service.py

```diff
@@ -1090,7 +1103,7 @@ def build(self, no_cache=False, pull=False, force_rm=False, memory=None, build_a
             forcerm=force_rm,
             pull=pull,
             nocache=no_cache,
-            dockerfile=build_opts.get('dockerfile', None),
```

Comment:
When the path is a URL or gzip is true, `APIClient` passes dockerfile through to the server, so this should probably still be provided in those cases